### PR TITLE
Fix uncache option for list rules and categories

### PR DESCRIPTION
**Pull request by emcfarlane**
Ensures when ClientWithCacheRulesAndCategories option is not supplied the default uncached behaviour is respected. Each invocation of ListRules or ListCatorgies by default should not be cached.

### CODE REVIEW

File: internal/pkg/cache/singleton.go [hunk 1]
```diff
@@ -27,9 +27,7 @@ type Singleton[V any] struct {
 	get   func(context.Context) (V, error)
 	value V
 	err   error
-	// Storing a bool to not deal with generic zero/nil comparisons.
-	called bool
-	lock   sync.RWMutex
+	once  sync.Once
```
Comment (emcfarlane) [inline internal/pkg/cache/singleton.go]: Unrelated change. The singleton can be written as a once.

Comment (bufdev) [reply internal/pkg/cache/singleton.go]: Undo this

Comment (emcfarlane) [reply internal/pkg/cache/singleton.go]: Reverted.

**Review by bufdev — APPROVED** [review]
